asserts/database.go

```diff
@@ -22,12 +22,42 @@
 package asserts
 
 import (
-	"errors"
 	"fmt"
 	"regexp"
 	"time"
 )
 
+// NotFoundError is returned when an assertion can not be found.
+type NotFoundError struct {
+	Type    *AssertionType
+	Headers map[string]string
+}
+
+// NewNotFoundErrorPrimaryKey builds a NotFoundError for a unique assertion of given assertType and primaryKey.
+func NewNotFoundErrorPrimaryKey(assertType *AssertionType, primaryKey []string) *NotFoundError {
+	headers, err := headersFromPrimaryKey(assertType, primaryKey)
+	if err != nil {
+		panic(err)
+	}
+	return &NotFoundError{Type: assertType, Headers: headers}
+}
+
+func (e *NotFoundError) Error() string {
+	pk, err := primaryKeyFromHeaders(e.Type, e.Headers)
+	if err != nil || len(e.Headers) != len(pk) {
+		// TODO: worth conveying more information?
+		return fmt.Sprintf("%s(s) not found", e.Type.Name)
```

Comment:
Perhaps rewording slightly so its clarity doesn't depend on context:

_("%s assertion not found", e.Type.Name)_

or this, if we want to show more details:

_("%s assertion not found for %s", e.Type.Name, headers)_

where headers might be formatted similar to _foo=bar, baz=bom_